### ISSUE TITLE
Unhandled event: cljs.core.PersistentVector

### ISSUE THREAD

**Issue by sova** [ptaoussanis/sente#288]
Hello,  I am getting strange errors on the client-side.

Using the example application as a scaffold, I've been trying to add my own events...

Here is some of my javascript console:
```
Channel socket successfully established!: cljs.core.PersistentArrayMap
client.cljs:159 Handshake: cljs.core.PersistentVector
client.cljs:319 Submitting testtext
client.cljs:101 Unhandled event: cljs.core.PersistentVector
```

As you can see, it does not actually decipher the PersistentVector, even though I am sending the name of an event like so:

server.clj
```
(chsk-send! uid [:serversent/blurb {:link "This is the blurb title"
                                          :content "Content"
                                          :author "Author"
                                          :total-score 159
                                          :number-of-ratings 2
                                          }])
```

client.cljs
```
; events

(defmethod -event-msg-handler
  :serversent/blurb ; Serversent blurb event
  [{:as ev-msg :keys [event]}]
  (.log js/console "Serversent/blurb: %s" event)
  )

(defmethod -event-msg-handler
  :default ; Default/fallback case (no other matching handler)
  [{:as ev-msg :keys [event]}]
  (.log js/console "Unhandled event: %s" event)
  )

```

Always falls back to :default because it doesn't make sense of the PersistentVector..

Any help is appreciated.

**Comment by danielcompton** [comment]
Does this work with the default application?

**Comment by sova** [comment]
@danielcompton yes

I have resolved the issue, but I had to change my clientside code to function in the following way:


(defn event-msg-handler  ;;could be enhanced by being made non-blocking
  [{:keys [_ _ ?data]}]
  (.log js/console (str "&&" ?data ))
  (.log js/console (str "&! " (first ?data)))
  (let [event-title (first ?data)
        new-data (second ?data)]
    (cond
      (= event-title :hello/client)
        (.log js/console (str "&# " new-data))
      (= event-title :serversent/blurb)
        (.log js/console (str "&# " new-data)))))


Which will display usefully to the js console 

&& Event-name+Payload
&!  Event-name
&# Payload


So I switched from using a defmulti to a defn with cond.  I believe that having both the defn and defmulti was doing something bizarre, but that is just an inclination.  Just to reiterate, it's resolved now.
